src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -866,4 +942,46 @@ public BuildRule getBinaryBuildRule() {
     }
     return ImmutableSortedSet.of();
   }
+
+
+  @Override
+  public Iterable<? extends NativeLinkable> getNativeLinkableDeps(CxxPlatform cxxPlatform) {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public Iterable<? extends NativeLinkable> getNativeLinkableExportedDeps(CxxPlatform cxxPlatform) {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public NativeLinkableInput getNativeLinkableInput(
+      CxxPlatform cxxPlatform, Linker.LinkableDepType type) throws NoSuchBuildTargetException {
+    ImmutableList.Builder<Arg> linkerArgsBuilder = ImmutableList.builder();
+    linkerArgsBuilder.addAll(StringArg.from(
```

Comment:
These rpaths should be setup differently depending on what type of bundle the binary ends up in. For iOS and for framework bundles it needs to be `@loader_path/Frameworks`, but for OSX bundles `@loader_path/../Frameworks`. I think this might not be the best place to add this, and that it could be more appropriate as a flavor on the binary. Including frameworks a bundle that contains a binary influences that binary's linker flags because these rpaths should be set or not (although I'm not sure if its bad if they're always set). The flavor on the binary would add the appropriate linker flags for the platform/bundle combination.

Thoughts?